modules/stochastic_tools/src/actions/AdaptiveSamplerAction.C

```diff
@@ -0,0 +1,70 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "AdaptiveSamplerAction.h"
+#include "AddSamplerAction.h"
+#include "FEProblem.h"
+
+registerMooseAction("StochasticToolsApp", AdaptiveSamplerAction, "add_user_object");
+registerMooseAction("StochasticToolsApp", AdaptiveSamplerAction, "add_postprocessor");
+
+InputParameters
+AdaptiveSamplerAction::validParams()
+{
+  InputParameters params = Action::validParams();
+  params.addClassDescription("Adds extra objects pertaining to adaptive samplers.");
+  return params;
+}
+
+AdaptiveSamplerAction::AdaptiveSamplerAction(const InputParameters & params) : Action(params) {}
+
+void
+AdaptiveSamplerAction::act()
+{
+  const auto samplers = _app.actionWarehouse().getActions<AddSamplerAction>();
+  SamplerName sampler_name;
+
+  for (auto & sampler : samplers)
+  {
+    std::string sampler_type = sampler->getMooseObjectType();
+
+    // determine whether the sampler is an adaptive sampler
+    auto adaptive_sampler_names = adaptiveSamplerNames();
+    for (auto & adaptive_sampler : adaptive_sampler_names)
+      if (sampler_type.compare(adaptive_sampler) == 0)
+      {
+        if (!sampler_name.empty())
+          mooseError("The case with multiple adaptive samplers is not currently supported");
+
+        sampler_name = sampler->name();
+      }
+  }
```

Comment:
Take or leave this suggestion, just a little leaner.

```suggestion
  for (auto & sampler : samplers)
    if (adaptiveSamplerNames().count(sampler->getMooseObjectType()) > 0)
    {
      if (!sampler_name.empty())
        mooseError("The case with multiple adaptive samplers is not currently supported");
      sampler_name = sampler->name();
    }
```